### PR TITLE
refactor: display solutions

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This lays the groundwork for https://github.com/freeCodeCamp/freeCodeCamp/issues/44979 by consolidating the code for displaying user submissions.  

I also fixed a bug where legacy JavaScript submissions (which were stored as `challenge.solution` not `challenge.files`) were not being displayed on the user's timeline.  

Finally, the case where the submission has neither a `solution` nor `files` is handled just by adding a suggestion to email support.  The rationale was that this shouldn't happen, but if it does it'll only be fixable by updating the database.

### CODE REVIEW

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
When opening the timeline to view a submitted Take Home Project - something with two links to submit.

```console
Warning: Failed prop type: The prop `id` is required to make `Dropdown` accessible for users of assistive technologies such as screen readers.
    in Dropdown (created by Uncontrolled(Dropdown))
    in Uncontrolled(Dropdown) (created by DropdownButton)
    in DropdownButton (created by SolutionDisplayWidget)
    in div (created by SolutionDisplayWidget)
    in SolutionDisplayWidget (created by TimelineInner)
```

File: client/src/components/solution-display-widget/index.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      {t('buttons.show-code')}
+    </Button>
+  );
+  const ShowProjectAndGithubLinks = (
+    <div className='solutions-dropdown'>
+      <DropdownButton
+        block={true}
+        bsStyle='primary'
+        className='btn-invert'
+        title={dropdownTitle}
```
Comment (ShaunSHamilton) [inline client/src/components/solution-display-widget/index.tsx:84]:
```suggestion
        className='btn-invert'
        id={'drop-down-${dropdownTitle}`}
        title={dropdownTitle}
```

Comment (ojeytonwilliams) [reply client/src/components/solution-display-widget/index.tsx:84]: Oh, bootstrap :man_facepalming:

**Review by ShaunSHamilton — APPROVED** [review]
Sure 👍

**Review by naomi-lgbt — APPROVED** [review]
looks good to me

**Pull request merged**